server/src/main/java/org/apache/kafka/server/purgatory/DelayedDeleteRecords.java

```diff
@@ -91,7 +91,13 @@ public boolean tryComplete() {
             }
         });
         //  check if every partition has satisfied at least one of case A or B
-        return deleteRecordsStatus.values().stream().noneMatch(DeleteRecordsPartitionStatus::acksPending) && forceComplete();
+        boolean noPendingPartition = deleteRecordsStatus.values()
+                .stream()
+                .noneMatch(DeleteRecordsPartitionStatus::acksPending);
+        if (noPendingPartition) {
+            forceComplete();
+        }
+        return noPendingPartition && isCompleted();
```

Comment:
if `noPendingPartition` is true then `forceComplete` will be invoked which should always return `isCompleted()` as true, correct? Hence the code shouldn't be like below?

 ```suggestion
        if (noPendingPartition) {
            forceComplete();
            return true;
        }
        return false;
```